### PR TITLE
FIX: Support working directories containing whitespace

### PR DESCRIPTION
**Pull request by osteensco**
addresses #546 #472, probably more

I believe this solution is OS agnostic but happy to adjust if it doesn't work somewhere.

### CODE REVIEW

**Comment by osteensco** [comment]
sorry for the messy merges... I was having issues with some of the tests that seemed unrelated to my changes, and the build process on github kept failing so I tried to catchup since I saw there was a linter problem fixed. I ended up just testing my PR manually on a local project and it works on my machine 🤷‍♂️ . Is there a known issue with some of the tests in their current state? I couldn't find any open issues about it.

**Comment by omarattia3143** [comment]
this is causing issue #659, I am not sure why does space in path an cause issues, test paths with spaces using windows 11 it works fine, so I am not sure how to test this but could you use something like this?
`if strings.Contains(c.Build.Bin, " ") && !strings.HasPrefix(c.Build.Bin, "\"") && !strings.HasSuffix(c.Build.Bin, "\"") {
		c.Build.Bin = "\"" + c.Build.Bin + "\""
	}`

**Comment by cosmtrek** [comment]
@omarattia3143 Sorry for the bug, I don't have windows for testing, PR is welcome~

**Comment by omarattia3143** [comment]
@osteensco can you confirm that this PR still works with this code? I have only windows and I can not test your changes

> this is causing issue #659, I am not sure why does space in path an cause issues, test paths with spaces using windows 11 it works fine, so I am not sure how to test this but could you use something like this? `if strings.Contains(c.Build.Bin, " ") && !strings.HasPrefix(c.Build.Bin, "\"") && !strings.HasSuffix(c.Build.Bin, "\"") { c.Build.Bin = "\"" + c.Build.Bin + "\"" }`

File: runner/config.go [hunk 1]
```diff
@@ -327,6 +329,9 @@ func (c *Config) preprocess() error {
 	// CMD will not recognize relative path like ./tmp/server
 	c.Build.Bin, err = filepath.Abs(c.Build.Bin)
 
+	// Account for spaces in filepath
+	c.Build.Bin = fmt.Sprintf("%q", c.Build.Bin)
```
Comment (JasonGoemaat) [inline runner/config.go:333]: I think this is causing an error on Windows, the exe can't be found and the path includes escape characters like this:

```
'\"c:\\go\\hello\\tmp\\hello.exe\"' is not recognized as an internal or external command,
operable program or batch file.
```

When if it was actually missing the error would be: this when the exe is missing and I run it from a command prompt
```
'c:\go\hello\tmp\hello.exe' is not recognized as an internal or external command,
operable program or batch file.
```

Comment (moboqe) [reply runner/config.go:333]: yep, that is not working properly

**Comment by moboqe** [comment]
> this is causing issue #659, I am not sure why does space in path an cause issues, test paths with spaces using windows 11 it works fine, so I am not sure how to test this but could you use something like this? `if strings.Contains(c.Build.Bin, " ") && !strings.HasPrefix(c.Build.Bin, "\"") && !strings.HasSuffix(c.Build.Bin, "\"") { c.Build.Bin = "\"" + c.Build.Bin + "\"" }`

That seems not working either

**Comment by omarattia3143** [comment]
@moboqe this is not merged, It works if you dont have space in your path, I made it so it can bypass the issue created by this PR.
I am not sure if this PR works for other platforms.
@cosmtrek I think the best is to revert this PR since the author is not responding and it is affecting all windows users

**Comment by moboqe** [comment]
> @moboqe this is not merged, It works if you dont have space in your path, I made it so it can bypass the issue created by this PR. I am not sure if this PR works for other platforms. @cosmtrek I think the best is to revert this PR since the author is not responding and it is affecting all windows users

I mean this doesn't work for me related for **this** issue.
Yep, I also fixed those issue by adding check for space in path

**Pull request merged**